contrib/bitcoin-mock.md

```diff
@@ -0,0 +1,138 @@
+# Run bitcoind regtest node
+
+## Requirements
+
+- Docker Engine >= 27.0
+- docker-compose-plugin (don't use the legacy python docker-compose)
+- docker service running; [install instructions](https://docs.docker.com/engine/install/).
+
+There are two leading Bitcoin implementation, each comes with a mode to run simulations that helps to use it
+as a mock, with predefined snapshot and data:
+
+- [Bitcoind](https://github.com/bitcoin/bitcoin) with the `-regtest` mode
+- [Btcd](https://github.com/btcsuite/btcd) with the `-simtest` mode
+
+## Bitcoind
+
+Bitcoind is a binary from the reference implementation
+
+### Setting up
+
+Firstly we need to copy the snapshot (to have a shared data):
+
+```sh
+make bitcoind-init
+```
+
+Start a new terminal where you will run a container with bitcoind, by default the nativewallet will be loaded:
+
+```sh
+cd contrib; docker compose up
+```
+
+To stop the node just press `ctrl-c` in the terminal running a node.
+
+To remove a container:
+
+```sh
+docker compose down
+```
+
+### Interact with the bitcoind node
+
+```sh
+docker exec -it bitcoind-node bitcoin-cli -regtest <args>
+```
+
+Or enter bash in the container:
+
+```sh
+docker exec -it bitcoind-node /bin/bash
+```
+
+Then you can generate a block:
+
+```sh
+bitcoin-cli -regtest generate <number-block>
+```
+
+If RPC params are required, you can provide them:
+
+```sh
+bitcoin-cli -regtest -rpcuser=user -rpcpassword=password generate
+```
```

Comment:
This is good if we can add one example use getblockchaininfo use bitcoin cli and curl